indico/util/user.py

```diff
@@ -181,3 +189,44 @@ def wrapper(*args, **kwargs):
         return fn(*args, **kwargs)
 
     return wrapper
+
+
+def create_user(user_form, handler, pending_user=None, is_admin=False):
```

Comment:
- Don't pass the form here, pass a data dict.
- Ensure that moderated registration also works fine in the external-login flow (e.g. when using oauth to login)
- do `handler.get_all_emails()` before creating the request - approving it should only use data from the request
- check for a `pending_user` when approving a registration request (same logic as during normal registration)